docs/sdk/java.md

```diff
@@ -364,6 +364,43 @@ PersistedTable<String, Integer> table = PersistedTable.of("my-table", String.cla
 PersistedAppendingBuffer<Integer> buffer = PersistedAppendingBuffer.of("my-buffer", Integer.class);
 {% endhighlight %}
 
+### Dynamic State Registration
+
+Using the above state types, a function's persisted state must be defined eagerly. You cannot use those state types to
+register new persisted state during invocations (i.e. in the ``invoke`` method) or after the function instance is created.
+
+If dynamic state registration is required, you can use a ``PersistedStateRegistry`` to achieve that:
```

Comment:
I can't exactly explain why, but the way you wrote it sounds off to me. 
```suggestion
If dynamic state registration is required, it can be achieved using a ``PersistedStateRegistry``:
```